web/lib/src/helpers/extensions.dart

```diff
@@ -103,3 +103,30 @@ extension TouchListConvert on TouchList {
   @Deprecated('Use JSImmutableListWrapper<TouchList, Touch> directly instead.')
   List<Touch> toList() => JSImmutableListWrapper<TouchList, Touch>(this);
 }
+
+extension XMLHttpRequestGlue on XMLHttpRequest {
+  Map<String, String> get responseHeaders {
+    // from Closure's goog.net.Xhrio.getResponseHeaders.
+    final headers = <String, String>{};
+    final headersString = getAllResponseHeaders();
+    final headersList = headersString.split('\r\n');
```

Comment:
You could use `ListSplitter.split` here. And get a lazy iterator...

Reply:
I can't find a reference to `ListSplitter` anywhere - could you provide a link?